src/halmos/sevm.py

```diff
@@ -2609,12 +2609,14 @@ def jumpi(
             follow_false = visited[False] < self.options.loop
             if not (follow_true and follow_false):
                 self.logs.bounded_loops.append(jid)
-                debug(
-                    f"\nloop id: {jid}\n"
-                    f"loop condition: {cond}\n"
-                    f"calldata: {ex.calldata()}\n"
-                    f"path condition:\n{ex.path}\n"
-                )
+                if self.options.debug:
+                    # rendering ex.path to string can be expensive, so only do it if debug is enabled
+                    debug(
+                        f"\nloop id: {jid}\n"
+                        f"loop condition: {cond}\n"
+                        f"calldata: {ex.calldata()}\n"
+                        f"path condition:\n{ex.path}\n"
+                    )
```

Comment:
oh wow. great catch. (curious why it wasn't caught earlier by my performance regression testing.)

Reply:
it has a disproportionate effect on the morpho test because it bumps against the loop limit and also has very complex path conditions, not much visible effect on other tests